bus-mapping/src/evm/opcodes/create.rs

```diff
@@ -159,23 +170,15 @@ impl<const IS_CREATE2: bool> Opcode for Create<IS_CREATE2> {
             state.call_context_write(&mut exec_step, caller.call_id, field, value);
         }
 
+        let (initialization_code, code_hash) = if is_precheck_ok && length > 0 {
```

Comment:
I guess here we missed one condition `!is_address_collision`. In circuit, we check `is_precheck_ok && not_address_collision` (https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1657/files#diff-66e8458ad6b55447e17de7f715e4395be943d682483be4a945b51f98a4cc50aeR261)

Reply:
i would suggest do cb.copy_table_lookup only with condition of is_precheck_ok but not `is_precheck_ok && not_address_collision`? The reason is a bit subtle like this:

Backgound: for normal create2 case, (a new contract is deployed), we should constrain the codehash (which is used to calculate the new callee.address). Currently we do this constrain implicitly using CopyTable with Bytecode as dst, and all bytecodes inside byecode circuit have correct keccak lookup.

For the create2 op, if precheck is ok, we need to hash the initcode to calculate the new address. If there is a collision, then in fact no initcode will be executed so for this collision case, only keccak lookup is needed while copy lookup with bytecode is not needed.

So we have 2 solutions here:
(1) do cb.copy_table_lookup only with condition of is_precheck_ok but not `is_precheck_ok && not_address_collision`. This method, code will be simpler. The only drawback is for collision case, the row usage of bytecode circuit is not optimal. (But i think it is generally ok, since collision is not very common).
(2) for happy case (initcode executed), we still do a single copy lookup with bytecode dst. But for "precheck is ok but collision" case, we need to change to (a) an copy lookup with RlcAcc instead of Bytecode as dst + (b) a keccak lookup. This method can save some bytecode circuit row usage, but make code very complex.

I would prefer solution (1). Any ideas on this?

Reply:
Solution (1) works for me and it does make the code simpler. If so, I think you need to change the condition from `is_precheck_ok && not_address_collision` to `is_precheck_ok` only in the circuit [here](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1657/files#diff-66e8458ad6b55447e17de7f715e4395be943d682483be4a945b51f98a4cc50aeR261).

Reply:
Besides, the reason behind is a bit subtle, could you add some comment as well?